### PR TITLE
added new functions

### PR DESCRIPTION
**Pull request by sillyElvy**
## About

This PR adds (again) new functions

## Checklist

- [x] I've ran `pre-commit` to format and lint the change(s) made.
- [x] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [x] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent): #415 
- [ ] I've made this pull request for/as: (check all that apply)
  - [x] Documentation
  - [ ] Breaking change
  - [x] New feature/enhancement
  - [ ] Bugfix

### CODE REVIEW

File: interactions/api/http.py [hunk 1]
```diff
@@ -569,7 +569,9 @@ async def modify_guild(
         :param reason: Reason to send to the audit log, if given.
         """
```
Comment (i0bs) [inline interactions/api/http.py:569]: Please include `return` and `rtype`

File: interactions/api/models/guild.py [hunk 2]
```diff
@@ -11,6 +12,44 @@
...
+
+class DefaultMessageNotificationLevel(IntEnum):
+    """An enumerable object representing the default message notification level of a guild."""
+
+    ALL_MESSAGES = 0
+    ONLY_MENTIONS = 1
+
+
+class EntityType(IntEnum):
+    STAGE_INSTANCE = 1
```
Comment (i0bs) [inline interactions/api/models/guild.py:43]:
```suggestion
    """An enumerable object representing the type of event."""
    STAGE_INSTANCE = 1
```

File: interactions/api/models/guild.py [hunk 3]
```diff
@@ -11,6 +12,44 @@
...
+    VOICE = 2
+    EXTERNAL = 3
+
+
+class EventStatus(IntEnum):
+    SCHEDULED = 1
```
Comment (i0bs) [inline interactions/api/models/guild.py:51]:
```suggestion
    """An enumerable object representing the status of an event."""
    SCHEDULED = 1
```

File: interactions/api/models/guild.py [hunk 4]
```diff
@@ -714,6 +753,283 @@ async def modify_member(
...
 
+    async def get_preview(self) -> "GuildPreview":
+        """
+        Get the guild's preview.
+        """
+        return GuildPreview(**await self._client.get_guild_preview(guild_id=int(self.id)))
+
+    async def leave(self) -> None:
+        """
+        Removes the bot from the guild.
```
Comment (i0bs) [inline interactions/api/models/guild.py]: I'm not sure if black formatted it (ignore if it did) but I believe the docstring should all be on one line for singular comments.

File: interactions/api/models/guild.py [hunk 5]
```diff
@@ -714,6 +753,283 @@ async def modify_member(
...
+        :type public_updates_channel_id: Optional[int]
+        :param preferred_locale?: The preferred locale of a community guild used in server discovery and notices from Discord; defaults to "en-US"
+        :type preferred_locale: Optional[str]
+        :param description?: The description for the guild, if the guild is discoverable
+        :type description: Optional[str]
+        :param premium_progress_bar_enabled?: Whether the guild's boost progress bar is enabled
+        :type premium_progress_bar_enabled: Optional[bool]
+        :param reason?: The reason for the modifying
+        :type reason: Optional[str]
+        """
```
Comment (i0bs) [inline interactions/api/models/guild.py:834]: Please include `return` and `rtype`. Refer to the docstring variance from `send` in `CommandContext` to see how that appears.

File: interactions/api/models/guild.py [hunk 6]
```diff
@@ -714,6 +753,283 @@ async def modify_member(
...
+        :type scheduled_start_time: datetime.isoformat
+        :param scheduled_end_time?: The time when the scheduled event is scheduled to end
+        :type scheduled_end_time: Optional[datetime.isoformat]
+        :param entity_metadata?: The entity metadata of the scheduled event
+        :type entity_metadata: Optional[EventMetadata]
+        :param channel_id?: The channel id of the scheduled event.
+        :type channel_id: Optional[int]
+        :param description?: The description of the scheduled event
+        :type description: Optional[str]
+        """
```
Comment (i0bs) [inline interactions/api/models/guild.py:927]: Please include `return` and `rtype`.

File: interactions/api/models/guild.py [hunk 7]
```diff
@@ -714,6 +753,283 @@ async def modify_member(
...
+        :param scheduled_start_time: The time to schedule the scheduled event
+        :type scheduled_start_time: Optional[datetime.isoformat]
+        :param scheduled_end_time?: The time when the scheduled event is scheduled to end
+        :type scheduled_end_time: Optional[datetime.isoformat]
+        :param entity_metadata?: The entity metadata of the scheduled event
+        :type entity_metadata: Optional[EventMetadata]
+        :param channel_id?: The channel id of the scheduled event.
+        :type channel_id: Optional[int]
+        :param description?: The description of the scheduled event
+        :type description: Optional[str]
```
Comment (i0bs) [inline interactions/api/models/guild.py:989]: Please include `return` and `rtype`.

**Review by i0bs — CHANGES_REQUESTED** [review]

File: interactions/api/models/guild.py (hunk 4 above)
Comment (sillyElvy) [reply interactions/api/models/guild.py]: Black ignores it in both cases

**Review by i0bs — APPROVED** [review]

**Pull request merged**